File: projects/order-creation/core/src/main/java/com/robinfood/core/entities/FinalProductEntity.java
package com.robinfood.core.entities;

import java.math.BigDecimal;
import java.util.List;
import lombok.Data;

@Data
public class FinalProductEntity {

    private final List<GroupEntity> groups;

    private final Long id;

    private final BigDecimal price;

    private final Integer quantity;

    private final String sku;

    private final String type;
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/entities/transactionrequest/FinalProductEntity.kt
package com.robinfood.core.entities.transactionrequest

import java.math.BigDecimal

data class FinalProductEntity(
    val article: ArticleEntity,
    val brand: BrandEntity,
    val co2Total: BigDecimal,
    val discounts: List<DiscountEntity>,
    val groups: List<GroupEntity>,
    val id: Long,
    val image: String,
    val name: String,
    val price: BigDecimal,
    val quantity: Int,
    val removedPortions: List<RemovedPortionEntity>,
    val size: SizeEntity,
    val sku: String,
    val taxes: List<TaxEntity>,
    val totalPrice: BigDecimal
)
